apps/voting/app/src/script.js

```diff
@@ -169,12 +219,39 @@ async function startVote(state, { creator, metadata, voteId }) {
   }))
 }
 
+async function getAccountVotes({ connectedAccount = '', votes = [] }) {
+  const connectedAccountVotes = await Promise.all(
+    votes.map(({ voteId }) =>
+      getAccountSpecificVote({ connectedAccount, voteId })
+    )
+  )
+    .then(voteStates =>
+      voteStates.reduce((states, { voteId, voteType }) => {
+        states[voteId] = voteType
+        return states
+      }, {})
+    )
+    .catch(console.error)
+
+  return connectedAccountVotes
+}
+
+async function getAccountSpecificVote({ connectedAccount = '', voteId = '' }) {
```

Comment:
What do you think about renaming this to `getVoterState()` to mimic the contract call?